transaction-status/src/parse_token.rs

```diff
@@ -539,6 +588,30 @@ mod test {
             }
         );
 
+        // Test InitializeMint2
+        let initialize_mint_ix = initialize_mint2(
+            &spl_token::id(),
+            &convert_pubkey(keys[0]),
+            &convert_pubkey(keys[2]),
+            Some(&convert_pubkey(keys[3])),
+            2,
+        )
+        .unwrap();
+        let message = Message::new(&[initialize_mint_ix], None);
+        let compiled_instruction = convert_compiled_instruction(&message.instructions[0]);
+        assert_eq!(
+            parse_token(&compiled_instruction, &keys).unwrap(),
```

Comment:
Just an example of the new handling from #23067 :
```suggestion
        let initialize_mint_ix = initialize_mint2(
            &spl_token::id(), // TODO: replace with `program_id`
            &convert_pubkey(mint_pubkey),
            &convert_pubkey(mint_authority),
            Some(&convert_pubkey(freeze_authority)),
            2,
        )
        .unwrap();
        let message = make_coerced_message(initialize_mint_ix, program_id);
        let compiled_instruction = convert_compiled_instruction(&message.instructions[0]);
        assert_eq!(
            parse_token(&compiled_instruction, &convert_account_keys(&message)).unwrap(),
```